framework/src/modules/interoperability/mainchain/commands/initialize_message_recovery.ts

```diff
@@ -60,24 +60,29 @@ export class InitializeMessageRecoveryCommand extends BaseInteroperabilityComman
 			throw new Error('Chain is not registered.');
 		}
 
+		const terminatedAccountExists = await this.stores
+			.get(TerminatedStateStore)
+			.has(context, params.chainID);
+		if (!terminatedAccountExists) {
+			throw new Error('Terminated state account not present.');
+		}
+
 		const terminatedAccount = await this.stores
 			.get(TerminatedStateStore)
 			.get(context, params.chainID);
```

Comment:
Out of curiosity, what happens with `get` if the account does not exist? Does it throw an error?

Reply:
Because in that case this extra code would not be completely necessary I guess

Reply:
It does throw an error which we would have to catch; the message looks like "Specified key: xxxx does not exist". It looks cleaner to use a `has` without catching and returning the error message we want.